File: AndroidTai/app/src/main/java/com/tai/androidtai/dagger/modules/AndroidModule.java
package com.tai.androidtai.dagger.modules;

import android.content.Context;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class AndroidModule {
    private final Context mContext;

    public AndroidModule(Context context) {
        mContext = context;
    }

    @Provides
    @Singleton
    Context provideContext() {
        return mContext;
    }
}

File: AndroidTai/app/src/main/java/com/tai/androidtai/dagger/modules/AndroidModule.kt
package com.tai.androidtai.dagger.modules

import android.content.Context

import javax.inject.Singleton

import dagger.Module
import dagger.Provides

@Module
class AndroidModule(private val mContext: Context) {

    @Provides
    @Singleton
    internal fun provideContext(): Context {
        return mContext
    }
}
